frontend/coprs_frontend/coprs/views/groups_ns/groups_general.py

```diff
@@ -31,22 +41,18 @@ def activate_group(fas_group):
 
         if fas_group not in flask.g.user.user_teams:
             raise InsufficientRightsException(
-                "User '{}' doesn't have access to fas group {}"
+                "User '{}' doesn't have access to group {}"
                 .format(flask.g.user.username, fas_group))
 
         alias = form.name.data
-        group = UsersLogic.get_group_by_fas_name_or_create(
-            fas_group, alias)
-
-        db.session.add(group)
+        group = UsersLogic.create_group_by_fas_name(fas_group, alias)
+        # db.session.add(group) is done by UsersLogic.create_group_by_fas_name()
+        db.session.flush()
```

Comment:
I bet `flush()`  is not needed.  I'd drop the commented-out line with `db.session.add()` (NB adding the same object twice shouldn't hurt).

Reply:
You are right, `commit()` will unconditionally call `flush()` before commit

Reply:
Yeah, `add()` is redundant, I have no objections to removing it.

Reply:
> Yeah, `add()` is redundant, I have no objections to removing it.

I totally misunderstand you :)
Done